### PR TITLE
fix(frontend): fix row filter with pk diff order type

### PR DESCRIPTION
**Pull request by xxhZs**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://raw.githubusercontent.com/risingwavelabs/risingwave/17af8a747593ebdbfa826691daf75bdab7d14fa0/.github/contributor-license-agreement.txt).

## Checklist

- [ ] I have written necessary rustdoc comments.
- [ ]  I have added necessary unit tests and integration tests.
- [ ]  I have added test labels as necessary. 
- [ ]  I have added fuzzing tests or opened an issue to track them. 
- [ ]  My PR contains breaking changes. 
- [ ]  My PR changes performance-critical code, so I will run (micro) benchmarks and present the results. 
- [ ]  My PR contains critical fixes that are necessary to be merged into the latest release. 

## Documentation

- [ ]  My PR needs documentation updates. 

<details>
<summary><b>Release note</b></summary>

</details>

### CODE REVIEW

File: src/frontend/planner_test/tests/testdata/output/row_filter.yaml [hunk 1]
```diff
@@ -10,14 +10,14 @@
     select * from t where (v1,v2) > (1,2);
   batch_plan: |-
     BatchExchange { order: [], dist: Single }
-    └─BatchScan { table: t, columns: [t.v1, t.v2, t.v3], scan_ranges: [(t.v1, t.v2) > (Int32(1), Int32(2))], distribution: UpstreamHashShard(t.v1, t.v2, t.v3) }
+    └─BatchScan { table: t, columns: [t.v1, t.v2, t.v3], scan_ranges: [(t.v1, t.v2) >= (Int32(1), Int32(2))], distribution: UpstreamHashShard(t.v1, t.v2, t.v3) }
```
Comment (hzxa21) [inline src/frontend/planner_test/tests/testdata/output/row_filter.yaml:13]: In addition to planner test, let's add e2e tests for the issues fixed by this PR as well to verify the result is correct.

File: src/frontend/src/utils/condition.rs [hunk 2]
```diff
@@ -415,35 +415,61 @@ impl Condition {
...
                         }
+                        match order_type {
+                            Some(o) => {
+                                if o != column_order.order_type {
+                                    all_added = false;
+                                    break;
+                                }
+                            }
+                            None => order_type = Some(column_order.order_type),
+                        }
```
Comment (hzxa21) [inline src/frontend/src/utils/condition.rs:435]: I think the check here will pass if the PKs are all ordered desc:
```
create materialized view mv1 as select * from t1 order by v1 desc, v2 desc, v3 desc;
```

However, in this case, I think the optimization is not applicable.

Comment (xxhZs) [reply src/frontend/src/utils/condition.rs:435]: We just need to make sure that the order type of all the pk's is the same, in the row_seq_scan_exec will generate the appropriate scan according to the order type of the first pk to scan the data.
https://github.com/risingwavelabs/risingwave/blob/fe84f69e1611dad37f0ea97ff9dcdcbbaa972b63/src/batch/executors/src/executor/row_seq_scan.rs#L438

File: e2e_test/batch/basic/row_filter.slt.part [hunk 3]
```diff
@@ -0,0 +1,65 @@
+statement ok
+create table t1(v1 int, v2 int, v3 int);
+
+statement ok
+insert into t1 values( 1,1,1),( 1,2,1),( 1,2,2),(1,3,1),(1,3,2),(1,3,3);
```
Comment (hzxa21) [inline e2e_test/batch/basic/row_filter.slt.part:5]: nits: fmt

File: src/frontend/src/utils/condition.rs (hunk 2 above)
Comment (hzxa21) [reply src/frontend/src/utils/condition.rs:435]: Got it. That means when all pks are ordered desc, with the following scan range:
```
ExprType::GreaterThan => {
    (Bound::Included(pk_struct), Bound::Unbounded)
}
```

the storage scan will be done by scanning from `Bound::Unbounded` to `Bound::Included(pk_struct)`

**Review by hzxa21 — APPROVED** [review]
LGTM

**Review by chenzl25 — APPROVED** [review]
LGTM

**Pull request merged**